File: calculator/src/main/java/com/turlir/calculator/translator/NotationTranslator.java
package com.turlir.calculator.translator;

import com.turlir.calculator.converter.Member;

import java.util.Iterator;
import java.util.Queue;

/**
 * Сортирует части выражения в соответствии со своей нотацией
 */
public interface NotationTranslator {

    /**
     *
     * @param parent математическое выражение
     * @return очередь из токенов того же выражения в другой нотации
     */
    Queue<Member> translate(Iterator<Member> parent) throws Exception;

}

File: calculator/src/main/java/com/turlir/calculator/translator/NotationTranslator.kt
package com.turlir.calculator.translator

import com.turlir.calculator.converter.Member
import java.util.*

/**
 * Сортирует части выражения в соответствии со своей нотацией
 */
interface NotationTranslator {

    /**
     *
     * @param parent математическое выражение
     * @return очередь из токенов того же выражения в другой нотации
     */
    @Throws(Exception::class)
    fun translate(parent: Iterator<Member>): Queue<Member>

}
